controllers/metallb_controller_test.go

```diff
@@ -0,0 +1,56 @@
+package controllers
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/metallb/metallb-operator/test/consts"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("MetalLB Controller", func() {
+	Context("syncMetalLB", func() {
+		speakerImage := "test-speaker:latest"
+		controllerImage := "test-controller:latest"
+		BeforeEach(func() {
+			By("Setting the environment variables")
+			Expect(os.Setenv("SPEAKER_IMAGE", speakerImage)).To(Succeed())
+			Expect(os.Setenv("CONTROLLER_IMAGE", controllerImage)).To(Succeed())
+		})
+		It("Should create manifests with the correct values", func() {
+			By("Creating a Metallb resource")
+			metallb := &v1alpha1.Metallb{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "metallb",
+					Namespace: consts.MetallbNameSpace,
+				},
+			}
+			err := k8sClient.Create(context.Background(), metallb)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Validating that the variables were templated correctly")
+			controllerDeployment := &appsv1.Deployment{}
+			Eventually(func() error {
+				err = k8sClient.Get(context.Background(), types.NamespacedName{Name: consts.MetallbDeploymentName, Namespace: consts.MetallbNameSpace}, controllerDeployment)
+				return err
+			}, 5*time.Second, 500*time.Millisecond).ShouldNot((HaveOccurred()))
+
+			Expect(len(controllerDeployment.Spec.Template.Spec.Containers)).To(BeNumerically(">", 0))
+			Expect(controllerDeployment.Spec.Template.Spec.Containers[0].Image).To(Equal(controllerImage))
```

Comment:
is there a way to check if the container is in running state from the status field instead of the above ?

Reply:
nope, iiuc envtest only spins up etcd and apiserver, meaning that there aren't any containers/pods starting as a result of the deployment being created.
I just wanted to simply validate that the controller is rendering the manifests correctly.